pkg/distributor/distributor_test.go

```diff
@@ -2141,6 +2170,7 @@ type mockIngester struct {
 	client.IngesterClient
 	grpc_health_v1.HealthClient
 	happy      bool
+	errFail    error
```

Comment:
Sorry, now I come to read this again I can't see why this is called `errFail`. 
It looks like this is the error that will be returned if not happy - is that correct?
Logically a 4xx error ought to be associated with the request not the ingester.

Reply:
Yes, you are right. I was trying to use the same pattern we already have for the test globally, but as this can be a 4xx or a 5xx it does not necessarily mean an error. I changed to 'failureResponse'. Let me know if you have another suggestion.